src/psyclone/parse/module_manager.py

```diff
@@ -307,11 +330,14 @@ def sort_modules(module_dependencies):
             # Take a copy so we can modify the original set of dependencies:
             dependencies_copy = dependencies.copy()
             for dep in dependencies_copy:
-                if dep not in todo:
+                if dep in todo:
+                    continue
+                # Print a warning if this module is not supposed to be ignored
+                if dep not in self.ignores():
+                    # TODO 2120: allow a choice to abort or ignore.
                     print(f"Module '{module}' contains a dependency to "
```

Comment:
This message is only printed if we know that 'mod' depends on 'dep', but we do not have 'dep' in our list of modules. If we would ignore this here, the sorting later would not work (since this module can never have all its dependencies satisfied). A module that has no dependencies will not print this messages (since the outer loop `for dep in dependencies_copy` has an empty list).

Reply:
Oh I see. I thought I must be missing something.